### PR TITLE
Updates to `/client`

### PR DESCRIPTION
**Pull request by sioanis**
**What's being updated:** 
- [x] Removes material UI
- [x] Adds storybook
- [x] Adds typescript
- [x] Adds styled components
- [x] Integrate prettier
- [x] Iron out apollo client logic

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -6,28 +6,37 @@
...
-    "react-scripts": "3.3.0"
+    "react-scripts": "3.3.0",
+    "styled-components": "^5.0.0",
+    "typescript": "^3.7.5"
   },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
```
Comment (sioanis) [inline client/package.json]: We could add this if we want to remove `react-scripts`

File: client/package.json [hunk 2]
```diff
@@ -6,28 +6,38 @@
...
   "version": "0.1.0",
   "description": "Tripster React App",
-  "main": "index.js",
   "dependencies": {
     "@apollo/react-hooks": "^3.1.3",
-    "@material-ui/core": "^4.8.3",
-    "@material-ui/icons": "^4.5.1",
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.4.0",
+    "@testing-library/user-event": "^7.2.1",
```
Comment (sioanis) [inline client/package.json]: remove this

File: client/public/index.html [hunk 3]
```diff
@@ -3,7 +3,7 @@
   <head>
     <meta charset="utf-8" />
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
-    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width" />
```
Comment (sioanis) [inline client/public/index.html]: TODO: Write more about scaling for mobile.

File: client/src/components/Login/index.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import { Link, Redirect } from "react-router-dom";
+import gql from 'graphql-tag';
+import { useMutation } from '@apollo/react-hooks';
+
+const LOGIN = gql`
```
Comment (sioanis) [inline client/src/components/Login/index.tsx]: Will remove this logic to somewhere not in the component

File: README.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# ✈️ tripster [![lerna](https://img.shields.io/badge/maintained%20with-lerna-cc00ff.svg)](https://lerna.js.org/)
+# ☁️ tripster [![lerna](https://img.shields.io/badge/maintained%20with-lerna-cc00ff.svg)](https://lerna.js.org/)
```
Comment (michaelyuen) [inline README.md]: 🤔

File: client/package.json [hunk 6]
```diff
@@ -6,28 +6,45 @@
   ],
   "version": "0.1.0",
   "description": "Tripster React App",
-  "main": "index.js",
   "dependencies": {
     "@apollo/react-hooks": "^3.1.3",
-    "@material-ui/core": "^4.8.3",
-    "@material-ui/icons": "^4.5.1",
```
Comment (michaelyuen) [inline client/package.json]: 🎉

File: client/package.json [hunk 7]
```diff
@@ -6,28 +6,45 @@
...
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.4.0",
+    "@types/jest": "^24.9.0",
+    "@types/node": "^12.12.25",
+    "@types/react": "^16.9.17",
+    "@types/react-dom": "^16.9.4",
+    "@types/react-router-dom": "^5.1.3",
     "apollo-boost": "^0.4.7",
     "graphql": "^14.5.8",
+    "husky": "^4.2.1",
```
Comment (michaelyuen) [inline client/package.json]: I think some of these may be `devDependencies` if you're trying to be strict. It's always hard for me to identify the difference.

Comment (michaelyuen) [reply client/package.json]: Also where is the husky config?

File: client/src/App.test.tsx [hunk 8]
```diff
@@ -2,6 +2,6 @@ import React from 'react';
 import { render } from '@testing-library/react';
 import App from './App';
 
-test('render', () => {
+test('renders', () => {
   expect(render(<App />));
```
Comment (michaelyuen) [inline client/src/App.test.tsx]: 👀

File: client/src/components/Button/Button.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { action } from '@storybook/addon-actions';
+import { Button } from '@storybook/react/demo';
+
+export default {
+  title: 'Button',
+  component: Button,
+};
+
+export const Text = () => <Button onClick={action('clicked')}>Hello Button</Button>;
```
Comment (michaelyuen) [inline client/src/components/Button/Button.stories.tsx]: Does action get invoked right away when written this way? Versus `() => action()`

File: client/src/components/Login/index.tsx [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+  mutation login($email: String!, $password: String!) {
+    login(email: $email, password: $password)
+  }
+`;
+
+export default function Login() {
+  const [credentials, setCredentials] = useState({});
+  const [login, { data }] = useMutation(LOGIN);
```
Comment (michaelyuen) [inline client/src/components/Login/index.tsx]: What is data?

File: client/src/components/Login/index.tsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+  if (data &&
+        data.login !== 'The password is invalid or the user does not have a password.'
+          && data.login !== 'Too many unsuccessful login attempts. Please try again later.') return <Redirect to='/dashboard' />;
+
+  return (
+    <form onSubmit={e => {
+      e.preventDefault();
```
Comment (michaelyuen) [inline client/src/components/Login/index.tsx]: I'm not sure, but if you use the `novalidate` attribute maybe you don't need to preventDefault.

File: client/src/components/Login/index.tsx [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+      if (credentials) login({ variables: credentials });
+    }}>
+      <input
+        onChange={e => setCredentials({...credentials, email: e.target.value})}
+      />
+      <input
+        onChange={e => setCredentials({...credentials, password: e.target.value})}
```
Comment (michaelyuen) [inline client/src/components/Login/index.tsx]: Hm, do these work? You don't need a value attribute or anything else?

File: client/src/react-app-env.d.ts [hunk 13]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (michaelyuen) [inline client/src/react-app-env.d.ts]: What's this do?

File: package.json [hunk 14]
```diff
@@ -30,5 +30,8 @@
   },
   "devDependencies": {
     "lerna": "^3.20.2"
+  },
+  "dependencies": {
+    "react-toast-component": "^1.3.13"
```
Comment (michaelyuen) [inline package.json]: Is this the right place? 💩

File: client/src/schema/index.ts [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+export const typeDefs = gql`
+  type Query {
+    isLoggedIn: Boolean!
+  }
+
+  type Mutation {
+    login(email: String!, password: String!): String # login token
+    signup(email: String!, password: String!): String
+  }
+`;
```
Comment (michaelyuen) [inline client/src/schema/index.ts]: Looking at this schema, makes me wonder if there is any value for TS here? Definitely in the resolvers and dataSources, though.

File: client/src/App.test.tsx (hunk 8 above)
Comment (sioanis) [reply client/src/App.test.tsx]: Will follow up in another PR.

File: client/src/components/Login/index.tsx (hunk 10 above)
Comment (sioanis) [reply client/src/components/Login/index.tsx]: Will follow up and change this.

File: client/src/schema/index.ts (hunk 15 above)
Comment (sioanis) [reply client/src/schema/index.ts]: Hmm, maybe not but I'm trying to keep things consistent and all type checked.

File: client/src/react-app-env.d.ts (hunk 13 above)
Comment (sioanis) [reply client/src/react-app-env.d.ts]: Removed.

File: client/package.json (hunk 7 above)
Comment (sioanis) [reply client/package.json]: removed this

File: client/src/components/Button/Button.stories.tsx (hunk 9 above)
Comment (sioanis) [reply client/src/components/Button/Button.stories.tsx]: removed this

File: client/src/components/Login/index.tsx (hunk 11 above)
Comment (sioanis) [reply client/src/components/Login/index.tsx]: i might remove no validate and style the error handling

File: client/src/components/Login/index.tsx (hunk 12 above)
Comment (sioanis) [reply client/src/components/Login/index.tsx]: Not yet.

File: client/src/App.test.tsx [hunk 16]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import App from "./App";
+
+test("renders", () => {
+  render(<App />);
```
Comment (michaelyuen) [inline client/src/App.test.tsx]: 💩

**Pull request merged**